File: rb/features.rb
# IpinfoDeveloper SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module IpinfoDeveloperFeatures
  def self.make_feature(name)
    case name
    when "base"
      IpinfoDeveloperBaseFeature.new
    when "ratelimit"
      IpinfoDeveloperRatelimitFeature.new
    when "retry"
      IpinfoDeveloperRetryFeature.new
    when "test"
      IpinfoDeveloperTestFeature.new
    when "timeout"
      IpinfoDeveloperTimeoutFeature.new
    else
      IpinfoDeveloperBaseFeature.new
    end
  end
end

File: py/features.py
# IpinfoDeveloper SDK feature factory

from feature.base_feature import IpinfoDeveloperBaseFeature
from feature.test_feature import IpinfoDeveloperTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpinfoDeveloperBaseFeature(),
        "test": lambda: IpinfoDeveloperTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
